### ISSUE TITLE
[Feature Request] Version control

### ISSUE THREAD

**Issue by tunbridgep** [giuspen/cherrytree#516]
So, currently there are some use cases I consider "bad" about cherrytree, but they aren't outright bugs. Namely:

While I can version my notes using github or dropbox, I only really have control over the cherrytree file as a single blob, which is hard to manage. I consider it extremely important being able to backup and restore versions of notes, as I edit them often, and they contain some of the most crucial information regarding my life. This is especially important as my laptop has a tendency to overwrite dropbox files with older versions when it feels like it.

Since cherrytree has the ability to import from and export to certain text-based files, I was hoping it could be extended to also work with them natively, such as saving to/from a folder structure rather than a database in a transparent manner, using md files for its notes (this is an example, I know markdown isn't really supported right now, but it's ubiquity with other programs is awesome). I understand that this would most possibly result in a loss of functionality as certain file types likely don't have all the information required for all of cherrytree's features, but the version control and backup possibilities make it worth it in my opinion.

An alternative to this would be adding version control to each node as a function within cherrytree, but that seems really difficult and like it wouldn't be that useful.

I hope this makes sense, I tend to describe things in...odd....ways

**Comment by giuspen** [comment]
I understand the problem of version control and I agree that the xml format (ctd) is not ideal to visualise the diffs but even though it may happen in future, the present is focused on ensuring the survival of cherrytree porting it out of the current dead library

**Comment by tunbridgep** [comment]
Yeah I get that there are more pressing issues. This is more of a "nice to have"

I was actually using the sqlite format and had forgotten about XML entirely, so it was even worse for me as git literally saw it as a binary file. I've switched to XML and it's much better but still not quite ideal. I agree there are much more important things at stake.

If this feature ever goes ahead, I've got some good ideas for formats which would make things easier to use with other software, such as a folder of md files, which would be a total breeze to work with

**Comment by moonburnt** [comment]
its indeed could be extremely handy to have such "out-of-box" ability - zim already has git plugin for such purpose (OP may consider migration to it, in case this feature is really critical), for example (but, I believe, internal structure of their storage is completely different from cherrytree's)

**Comment by singularis-mzf** [comment]
I would also appreciate a version control feature, especially for syntax-highlited nodes. I think it could be easily implemented through generating hidden read-only subnodes for each „snapshot“ of the particular node and as an optimization diff-patches could be stored inside them instead of full text. (I don't need this functionality for formatted nodes, only for plain text and syntax-highlighted nodes.)

**Comment by quesada** [comment]
I would love this feature.
